src/demo/service/demo-dependency-editor.service.ts

```diff
@@ -0,0 +1,64 @@
+import { Injectable } from '@angular/core';
+import { Observable } from 'rxjs';
+
+import { Headers, Http, RequestOptions, Response } from '@angular/http';
+
+import { DependencyEditorService } from '../../app/launcher/launcher.module';
+
+import { HelperService } from '../../app/launcher/service/helper.service';
+import { TokenProvider } from '../../app/service/token-provider';
+
+@Injectable()
+export class DemoDependencyEditorService implements DependencyEditorService {
+
+  private END_POINT: string = '';
+  private API_BASE: string = 'booster-catalog/';
+  private ORIGIN: string = '';
+
+  constructor(
+    private http: Http,
+    private helperService: HelperService,
+    private tokenProvider: TokenProvider
+  ) {
+    if (this.helperService) {
+      this.END_POINT = this.helperService.getBackendUrl();
+      this.ORIGIN = this.helperService.getOrigin();
+    }
+  }
+
+  private get options(): Observable<RequestOptions> {
+    let headers = new Headers();
+    headers.append('X-App', this.ORIGIN);
+    return Observable.fromPromise(this.tokenProvider.token.then((token) => {
+      headers.append('Authorization', 'Bearer ' + token);
+      return new RequestOptions({
+          headers: headers
+      });
+    }));
+  }
+
+  getBoosterInfo(missionId: string, runtimeId: string, runtimeVersion: string): Observable<any> {
```

Comment:
Do we need this at two places https://github.com/fabric8-launcher/ngx-launcher/pull/210/files#diff-2ddfb45de75888efe378637bd68993edR120? Can it be removed from `src/demo/service/demo-dependency-check.service.ts`?